### ISSUE TITLE
Cannot reference closure in while loop with `task.unique`

### ISSUE THREAD

**Issue by tal** [custom-components/pyscript#24]
When I run the following script I get an error when I try to reference a closure defined `task.unique`.

```
Exception in <file.test_svc.test_svc> line 19:
        method_post_unique()
        ^
NameError: name 'method_post_unique' is not defined
```

```py
@service
def test_svc(entity_id=None):
  if entity_id is None:
    log.error(f"No entity id passed")
    return

  def method_pre_unique():
    return "pre-unique"

  log.debug(f"pre-unique for {entity_id}")
  task.unique(f"test_svc|{entity_id}")
  log.debug(f"post-unique for {entity_id}")

  def method_post_unique():
    return method_pre_unique() + " post-unique"

  cnt = 0
  def inc_and_test():
    method_post_unique()
    method_pre_unique()
    cnt += 1
    return cnt < 5

  while inc_and_test():
    log.debug(f"running loop {cnt} {method_post_unique()} {method_pre_unique()}")

    task.sleep(0.1)
```

**Comment by craigbarratt** [comment]
What versions are you running?

**Comment by tal** [comment]
Sorry, my versions are:
pyscript: 0.21
hacs: 1.6.0
hass: 0.115.2

I noticed even variables have to be passed into the `inc_and_test` method as was trying to mitigate this issue in the script I'm [working on right now](https://gist.github.com/tal/349e52777811724ef7420abe1c735b76)

**Comment by craigbarratt** [comment]
Closures have some bugs in v0.21.  Please try running the master version.

When I quickly tried your code, there is one bug that the master version correctly reports: `cnt` in `inc_and_test()` is not defined when it executes `cnt += 1`.  To modify a variable that is defined in an outer scope, you need to declare it as a `nonlocal`, eg:
```python
  cnt = 0
  def inc_and_test():
    nonlocal cnt
    method_post_unique()
    method_pre_unique()
    cnt += 1
    return cnt < 5
```

**Comment by tal** [comment]
Thanks! I've got to get to bed but I'll update to master and try out `nonlocal`. Sorry if that's standard, I'm new to python.

If you check the [code I linked above](https://gist.github.com/tal/349e52777811724ef7420abe1c735b76) I don't do any modification in the closure and I still have to pass in many of the variables. But only into the method called for the while loop.

Thanks for the awesome integration and fast help!
